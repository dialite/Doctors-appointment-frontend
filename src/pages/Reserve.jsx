import React from 'react';
import ReservationForm from '../components/reservation/ReservationForm';

export default function Reserve() {
  return (
    <ReservationForm />
  );
}
