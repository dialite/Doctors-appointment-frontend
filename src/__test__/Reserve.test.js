import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Reserve from '../pages/Reserve';
import store from '../redux/configureStore';

describe('Reserve', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Reserve />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('has a Reserve button', () => {
    const reserve = render(
      <Provider store={store}>
        <BrowserRouter>
          <Reserve />
        </BrowserRouter>
      </Provider>,
    );
    expect(reserve.findByText('Reserve')).toMatchSnapshot();
  });
});
