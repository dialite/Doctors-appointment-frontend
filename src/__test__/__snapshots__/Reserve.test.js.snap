// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reserve renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="reservation"
        >
          <div
            class="reservation-container"
          >
            <h2
              class="reservation-title"
              data-testid="title"
            >
              Make your reservations
            </h2>
            <h3>
              (*)only complete CITY and DATETIME fields
            </h3>
            <form
              class="addForm"
            >
              <label
                for="city"
              >
                <b>
                  City
                </b>
                <input
                  autocomplete="off"
                  id="city"
                  name="city"
                  placeholder="Enter City"
                  required=""
                  type="text"
                  value=""
                />
              </label>
              <label
                for="datetime"
              >
                <b>
                  Datetime
                </b>
                <input
                  autocomplete="off"
                  id="datetime"
                  name="datetime"
                  placeholder="Enter Datetime"
                  required=""
                  type="datetime-local"
                  value=""
                />
              </label>
              <label
                for="doctor"
              >
                <b>
                  Doctor
                </b>
                <input
                  autocomplete="off"
                  disabled=""
                  id="doctor"
                  name="doctor"
                  type="text"
                  value="Dr. undefined undefined"
                />
              </label>
              <label
                for="user"
              >
                <b>
                  Username
                </b>
                <input
                  autocomplete="off"
                  disabled=""
                  id="user"
                  name="user"
                  type="text"
                  value=""
                />
              </label>
              <input
                class="addButton"
                type="submit"
                value="Reserve"
              />
            </form>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <section
        class="reservation"
      >
        <div
          class="reservation-container"
        >
          <h2
            class="reservation-title"
            data-testid="title"
          >
            Make your reservations
          </h2>
          <h3>
            (*)only complete CITY and DATETIME fields
          </h3>
          <form
            class="addForm"
          >
            <label
              for="city"
            >
              <b>
                City
              </b>
              <input
                autocomplete="off"
                id="city"
                name="city"
                placeholder="Enter City"
                required=""
                type="text"
                value=""
              />
            </label>
            <label
              for="datetime"
            >
              <b>
                Datetime
              </b>
              <input
                autocomplete="off"
                id="datetime"
                name="datetime"
                placeholder="Enter Datetime"
                required=""
                type="datetime-local"
                value=""
              />
            </label>
            <label
              for="doctor"
            >
              <b>
                Doctor
              </b>
              <input
                autocomplete="off"
                disabled=""
                id="doctor"
                name="doctor"
                type="text"
                value="Dr. undefined undefined"
              />
            </label>
            <label
              for="user"
            >
              <b>
                Username
              </b>
              <input
                autocomplete="off"
                disabled=""
                id="user"
                name="user"
                type="text"
                value=""
              />
            </label>
            <input
              class="addButton"
              type="submit"
              value="Reserve"
            />
          </form>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
